### PR TITLE
os/se/ameba: Add signing verification of user image

### PR DESCRIPTION
**Pull request by ZhenBei-Sin**

### CODE REVIEW

File: os/se/ameba/rtl_se_crypto_function.h [hunk 1]
```diff
@@ -124,3 +124,22 @@ typedef struct {
 	hal_key_type factory_slot_key_type[USABLE_FACTORY_KEY_INDEX];
 } factory_struc;
 
+/* Kernel Image Verification Structure */
+#include "amebahp_secure_boot.h"
```
Comment (sunghan-chang) [inline os/se/ameba/rtl_se_crypto_function.h:128]: This causes build break on RTL8721CSM. Please find it on CI build test.

**Review by sunghan-chang — COMMENTED** [review]
Noone knows the TPdual in public.
How about below for title?
```
os/se/ameba: add signing verfication of user image
```

File: os/se/ameba/rtl_se_smccc_wrapper.c [hunk 2]
```diff
@@ -346,3 +346,25 @@ int ameba_hal_SignatureVerify(uint32_t *imagehdr_addr)
 	return ret;
 }
 
+/* Kernel Image Verification */
```
Comment (sunghan-chang) [inline os/se/ameba/rtl_se_smccc_wrapper.c:349]: This is a common file of ameba chips. Is this ok to add this without any chip config?

Comment (ZhenBei-Sin) [reply os/se/ameba/rtl_se_smccc_wrapper.c:349]: Hi @sunghan-chang ,

This file only will build for ameba smart.
image
The chio config was covered at the /os/se/ameba/Make.defs

File: os/se/ameba/rtl_se_crypto_function.h (hunk 1 above)
Comment (ZhenBei-Sin) [reply os/se/ameba/rtl_se_crypto_function.h:130]: Hi @sunghan-chang ,
Apologies for the inconvenience caused, added the CONFIG_AMEBASMART_TRUSTZONE.
Thank You.

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**